utils/cargo-gbuild/src/cli.rs

```diff
@@ -0,0 +1,145 @@
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+use crate::manifest;
+use anyhow::{anyhow, Result};
+use ccli::{clap, Parser};
+use gear_wasm_builder::{
+    optimize::{self, OptType, Optimizer},
+    CargoCommand,
+};
+use std::{
+    fs,
+    path::{Path, PathBuf},
+};
+
+const ARTIFACT_DIR: &str = "gbuild";
+
+/// Command `gbuild` as cargo extension.
+#[derive(Parser)]
+pub struct GBuild {
+    /// The path to the program manifest
+    #[clap(short, long, default_value = "Cargo.toml")]
+    pub manifest_path: PathBuf,
+
+    /// Space or comma separated list of features to activate
+    #[clap(short, long)]
+    pub features: Vec<String>,
+
+    /// If enables the release profile.
+    #[clap(short, long)]
+    pub release: bool,
+
+    /// Directory for all generated artifacts
+    ///
+    /// If not set, the default value will be the target folder
+    /// of the cargo project.
+    #[clap(short, long)]
+    pub target_dir: Option<PathBuf>,
+
+    /// TODO: If enable meta build (#3850)
+    #[clap(short, long)]
+    pub meta: bool,
+}
+
+impl GBuild {
+    // Collects the artifacts.
+    ///
+    /// TODO: support gtest (#3851)
+    pub fn collect(&self) -> Result<Artifact> {
+        let root = self
+            .manifest_path
+            .parent()
+            .ok_or_else(|| anyhow!("Failed to parse the root directory."))?;
+
+        let manifest =
+            toml::from_str::<manifest::Manifest>(&fs::read_to_string(&self.manifest_path)?)
+                .map_err(|e| anyhow!("Failed to find the manifest file, {e}"))?;
+        let cargo_target_dir = manifest
+            .build
+            .and_then(|b| b.target_dir)
+            .unwrap_or(root.join("target"));
+
+        // Register the path of the built artifacts.
+        let name = manifest.package.name.replace('-', "_");
+        let gbuild_target_dir = self
+            .target_dir
+            .clone()
+            .unwrap_or(cargo_target_dir.clone())
+            .join(ARTIFACT_DIR);
+
+        self.cargo(&cargo_target_dir)?;
+        let artifact = Artifact {
+            program: gbuild_target_dir.join(format!("{name}.wasm")),
+            // TODO: support meta build (#3850)
+            //
+            // target_dir.join(format!("{name}.meta.wasm")),
+            meta: None,
+            root: gbuild_target_dir,
+        };
+
+        // Optimize the built wasm and return the artifact
+        artifact.process(cargo_target_dir.join(format!(
+            "wasm32-unknown-unknown/{}/{name}.wasm",
+            if self.release { "release" } else { "debug" }
+        )))?;
+        Ok(artifact)
+    }
+
+    /// Process the cargo command.
+    ///
+    /// TODO: support workspace build. (#3852)
+    fn cargo(&self, target_dir: &Path) -> Result<()> {
+        let mut kargo = CargoCommand::default();
+        if self.release {
```

Comment:
Also we should add support for `--profile`

Reply:
https://github.com/gear-tech/gear/pull/3835/commits/539214d4bcf29fda941e205f4af611797229cb88